controller/frontend/src/Controller/Frontend/Product/Standard.php

```diff
@@ -100,9 +100,6 @@ public function addFilterCategory( \Aimeos\MW\Criteria\Iface $filter, $catId,
 			$catIds = $list;
 		}
 
-		$expr = array( $filter->compare( '==', 'index.catalog.id', array_unique( $catIds ) ) );
-		$expr[] = $filter->getConditions();
```

Comment:
Removing default conditions seems to be wrong

Reply:
I moved them into the `else` clause so that only one id condition is used in every case.

Reply:
It's not there but it would also be wrong because there may be other conditions already set in the filter

Reply:
Can you please elaborate?

I replaced the default "category ids" condition by a filter function which also filters for category ids. But the filter function respects the category type parameter.

IIUC the category id condition is not necessary with the filter function. If I'm wrong, I will be happy to restore it, but it seems to be a duplicate with the new solution.

Reply:
Replacing `index.catalog.id` by `index.catalog:position` is OK but you also removed `$expr[] = $filter->getConditions();` which is definitively wrong.

Reply:
Sorry, I misunderstood. Fixed it. Thanks!

Reply:
You should add `$expr = [$filter->getConditions()];` here and not at the bottom because with your current code, you have an uninitialized variable

Reply:
Agree, fixed. Okay like that?

Reply:
Can you add `$expr[] = $this->filter->compare( '==', 'index.catalog.id', $ids );` too, please? This is also in 2019.x because it's a much more performant filter to reduce the records before fitlering further with `index.catalog:position`

Reply:
As you like :-)

Reply:
One more bug: `$this->filter` -> `$filter`

Reply:
Found one more and fixed it, too. Also, I added the array_unique() at all places.